### PR TITLE
add a few encodings

### PR DESCRIPTION
**Pull request by dericed**

### CODE REVIEW

**Comment by bturkus** [comment]
super great addition! just tested, and everything seems to working well. few questions:
1) some parts of the pashua window are looking crunched for me, but I'm not sure if it's my computer. does it look like this for others?

screen shot 2018-11-27 at 3 01 47 pm

2) when I select h264/aac in mp4, I'm not ending up with a .mp4 file.  am I missing something?

**Comment by b00giehead** [comment]
Would it be possible to have a slices option for FFV1? It's set to 16 at the moment I see, but this archive has highlighted 24 as a preference for BFI supplied material, so it would be nice to be able to code straight to 24 using vrecord.  Thanks and apologies if this is in the wrong place! Joanna

**Comment by dericed** [comment]
Hi @b00giehead, that's a good idea. Do you mind starting a [new issue](https://github.com/amiaopensource/vrecord/issues/new) with the comment there.

**Comment by retokromer** [comment]
@dericed Do you mind adding also [HuffYUV](https://github.com/amiaopensource/vrecord/issues/366) as capture format?

**Comment by dericed** [comment]
@retokromer I'm unfamiliar/uncertain with huffyuv encoding strategies, but feel welcome to reopen https://github.com/amiaopensource/vrecord/issues/366 if it's still a wish.

**Comment by retokromer** [comment]
@dericed Just FYI: two of the vendors which provide now FFV1 use internally HuffYUV and convert to FFV1 after, because it’s faster. (See also my article in the _IASA Journal_, n. 49, p. 14–18. Sorry for quoting myself.)

**Comment by dericed** [comment]
PR is updated to new gui. It's also incomplete as the NTSC 486 line input with 4:2:0 h264 encoding somehow causes a "height not divisible by 4" even though 4:2:0 encoding should not require this.

**Comment by retokromer** [comment]
Can this be merged?

**Comment by retokromer** [comment]
@dericed What is missing? (I would like to add HuffYUV once merged, in order to avoid to have to resolve conflicts.)

**Comment by dericed** [comment]
@retokromer it needs an approving review from someone who has tested. @bturkus tested earlier but this was before the ui was refactored.

**Comment by retokromer** [comment]
ping @bturkus

**Comment by bturkus** [comment]
just tested again. prores standard and HQ both seem to be working properly, but h264/aac I'm getting the same "height not divisible" error: 

```
[Parsed_pan_2 @ 0x7f9f3c101ec0] Pure channel mapping detected: 0 1
[libx264 @ 0x7f9f3e80a000] height not divisible by 4 (720x486)0/0   
Error initializing output stream 0:0 -- Error while opening encoder for output stream #0:0 - maybe incorrect parameters such as bit_rate, rate, width or height
Conversion failed!
pipe:: Function not implemented

analyzing input file... -
invalid input, analyzing aborted.. 
 2019-02-13T11:19:16 -  [vrecord]Error: Running: /usr/local/opt/ffmpegdecklink/bin/ffmpeg-dl -nostdin -nostats -t 4500 -loglevel info -f decklink -draw_bars 0 -audio_input embedded -video_input sdi -format_code ntsc -channels 8 -audio_depth 32 -raw_format uyvy422 -timecode_format none -i UltraStudio HD Mini -metadata:s:v:0 encoder=H.264 -map_metadata 0:s:v:0 -color_primaries smpte170m -color_trc bt709 -colorspace smpte170m -color_range mpeg -metadata creation_time=now -movflags write_colr -c:v libx264 -flags +ilme+ildct -crf 18 -pix_fmt yuv420p -c:a aac -filter_complex [0:v:0]setsar=40/27,setdar=4/3; [0:a:0]pan=stereo| c0=c0 | c1=c1[stereo1] -map [stereo1] -f mp4 /Users/benjaminturkus/Desktop/vrecord_tests/h264.mp4 -an -f framemd5 /Users/benjaminturkus/Desktop/vrecord_tests/h264.framemd5 -c copy -map 0 -f matroska -write_crc32 0 -live true - gave an Error Code - 1
Consider reporting this or asking for help at https://github.com/amiaopensource/vrecord/issues
```

**Comment by dericed** [comment]
hmm, maybe for h264 we need to crop if ntsc, 4 lines off top and 2 from bottom to get to 480.

**Comment by retokromer** [comment]
Thank you for reporting!

**Comment by dericed** [comment]
going to resolve https://github.com/amiaopensource/vrecord/pull/417 before finalizing this one

**Comment by dericed** [comment]
pr is updated, but the h264 encoding is showing in mediainfo as mbaff/tff on ntsc bff input

**Comment by bturkus** [comment]
yeah im seeing the same thing. would st8 to progressive be bad bad?

**Comment by retokromer** [comment]
ping

**Comment by retokromer** [comment]
This one is open since more than one year…

**Comment by dericed** [comment]
this is rebased and now forces the h264 encoding through yadif to be progressive.

File: vrecord [hunk 1]
```diff
@@ -1139,10 +1163,10 @@ drawgrid=w=256:h=256:t=1:c=white@0.2"
 # list of selections for each vrecord option
 VIDEO_INPUT_OPTIONS=("Composite" "SDI" "Component" "S-Video")
 AUDIO_INPUT_OPTIONS=("Analog" "SDI Embedded Audio" "Digital Audio (AES/EBU)")
-CONTAINER_OPTIONS=("QuickTime" "Matroska" "AVI" "MXF")
-VIDEO_CODEC_OPTIONS=("Uncompressed Video" "FFV1 version 3" "JPEG2000" "ProRes" )
+CONTAINER_OPTIONS=("QuickTime" "Matroska" "AVI" "MXF" "MP4")
+VIDEO_CODEC_OPTIONS=("Uncompressed Video" "FFV1 version 3" "JPEG2000" "ProRes" "ProRes (HQ)" "h264")
 FFV1_SLICE_OPTIONS=("4" "6" "9" "12" "16" "24" "30")
-AUDIO_CODEC_OPTIONS=("24-bit PCM" "24-bit FLAC")
+AUDIO_CODEC_OPTIONS=("24-bit PCM" "24-bit FLAC" "aac")
```
Comment (privatezero) [inline vrecord]: Super petty, but shouldn't aac be AAC

**Comment by dericed** [comment]
thx @privatezero, i squashed the commit with x264opts and changed the case of 'aac'. please re-review

File: vrecord [hunk 2]
```diff
@@ -1139,10 +1163,10 @@ drawgrid=w=256:h=256:t=1:c=white@0.2"
 # list of selections for each vrecord option
 VIDEO_INPUT_OPTIONS=("Composite" "SDI" "Component" "S-Video")
 AUDIO_INPUT_OPTIONS=("Analog" "SDI Embedded Audio" "Digital Audio (AES/EBU)")
-CONTAINER_OPTIONS=("QuickTime" "Matroska" "AVI" "MXF")
-VIDEO_CODEC_OPTIONS=("Uncompressed Video" "FFV1 version 3" "JPEG2000" "ProRes" )
+CONTAINER_OPTIONS=("QuickTime" "Matroska" "AVI" "MXF" "MP4")
+VIDEO_CODEC_OPTIONS=("Uncompressed Video" "FFV1 version 3" "JPEG2000" "ProRes" "ProRes (HQ)" "h264")
```
Comment (retokromer) [inline vrecord]: Sorry for nit picking: should we not say `H.264` rather than `h264`? (Also at line 844.)

**Pull request merged**